src/io.rs

```diff
@@ -138,34 +162,156 @@ impl Read for File {
 
         Ok(len)
     }
+}
 
-    fn read_to_string(&mut self, str: &mut String) -> Result<usize, SimpleLinkError> {
-        let mut buf: Vec<u8> = Vec::new();
-        let len = try!(self.read_to_end(&mut buf));
+impl Write for File {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, SimpleLinkError> {
+        let len = try!(self.write_at(buf, self.offset));
+        self.offset += len;
+        Ok(len)
+    }
+}
 
-        match String::from_utf8(buf) {
-            Ok(res) => {
-                str.clear();
-                str.push_str(&res);
-            },
-            Err(_) => {
-                return Result::Err(SimpleLinkError::FileSystem(FileSystemError::NOT_SUPPORTED))
-            },
+impl Seek for File {
+    fn seek(&mut self, pos: u64) -> Result<u64, SimpleLinkError> {
+        self.offset = pos as usize;
+        Ok(self.offset as u64)
+    }
+}
+
+//
+// Image files
+//
+
+pub struct Image {
+    offset: usize,
+    file_handle: i32,
```

Comment:
Sure, but just declaring a type alias like:
```rust
type FileHandle = i32;
```
will hide the implementation details. No big deal anyway.